File: python/206_ReverseList.py
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
               1 > 2 > 3 > None
        None < 1 < 2 < 3
        """

        prev = None
        curr = head

        while curr is not None:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp
        return prev

File: golang/206_ReverseList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
package main
import (
    "fmt"
)


type ListNode struct {
    Val int
    Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
    var prev *ListNode
    curr := head

    for curr != nil {
        tmp := curr.Next
        curr.Next = prev
        prev = curr
        curr = tmp
    }
    return prev

}


func main() {
    ll := &ListNode{1, nil}
    ll.Next = &ListNode{2, nil}
    ll.Next = &ListNode{3, nil}
    fmt.Println(reverseList(ll))

}

